### PR TITLE
restrict video/:id/edit to channel owner

### PR DESCRIPTION
**Pull request by mnaamani**

### CODE REVIEW

**Comment by siman** [comment]
In the latest PR's (https://github.com/Joystream/apps/pull/297/commits/ad31dea82e4d73661ea501ef37ae6852dc0fc56d#diff-82c383620da9b7138790c90b5b7d7400R210) I implemented the loading of a channel on load video entity. So if an entity (that is video) was found, then there should be `channel` property inside. This means that you don't need to do an extra query the `transport`.

**Comment by mnaamani** [comment]
> In the latest PR's ([ad31dea#diff-82c383620da9b7138790c90b5b7d7400R210](https://github.com/Joystream/apps/commit/ad31dea82e4d73661ea501ef37ae6852dc0fc56d#diff-82c383620da9b7138790c90b5b7d7400R210)) I implemented the loading of a channel on load video entity. So if an entity (that is video) was found, then there should be `channel` property inside. This means that you don't need to do an extra query the `transport`.

okay reverted to use the property on entity

File: packages/joy-media/src/upload/UploadVideo.tsx [hunk 1]
```diff
@@ -296,7 +303,7 @@ const InnerForm = (props: MediaFormProps<OuterProps, FormValues>) => {
     // because the first 'EntityCreated' event corresponds to a Media Object Entity.
     const events = filterSubstrateEventsAndExtractData(txResult, 'EntityCreated')
 
-    // Return if there were less than two event: 
+    // Return if there were less than two event:
```
Comment (siman) [inline packages/joy-media/src/upload/UploadVideo.tsx]:
```suggestion
    // Return if there were less than two events:
```

File: packages/joy-media/src/upload/UploadVideo.tsx [hunk 2]
```diff
@@ -93,6 +96,10 @@ const InnerForm = (props: MediaFormProps<OuterProps, FormValues>) => {
     return <em>ERROR: "Video" entity class in undefined</em>
   }
 
+  if (entity && entity.channel && !entity.channel.roleAccount.eq(myAccountId)) {
+    return <em>ERROR: "Only owner can edit video"</em>
```
Comment (siman) [inline packages/joy-media/src/upload/UploadVideo.tsx]:
```suggestion
    return <em>ERROR: Only owner can edit video</em>
```

File: packages/joy-media/src/upload/UploadVideo.tsx [hunk 3]
```diff
@@ -93,6 +96,10 @@ const InnerForm = (props: MediaFormProps<OuterProps, FormValues>) => {
     return <em>ERROR: "Video" entity class in undefined</em>
```
Comment (siman) [inline packages/joy-media/src/upload/UploadVideo.tsx]:
```suggestion
    return <em>ERROR: Video entity class in undefined</em>
```

File: packages/joy-media/src/upload/UploadVideo.tsx [hunk 4]
```diff
@@ -93,6 +96,10 @@ const InnerForm = (props: MediaFormProps<OuterProps, FormValues>) => {
...
   }
 
+  if (entity && entity.channel && !entity.channel.roleAccount.eq(myAccountId)) {
```
Comment (siman) [inline packages/joy-media/src/upload/UploadVideo.tsx]: There is a utility function `isAccountAChannelOwner`, please use it instead. Don't forget the import :)

`import { isAccountAChannelOwner } from '../channels/ChannelHelpers'`

```suggestion
  if (entity && !isAccountAChannelOwner(entity.channel, myAccountId)) {
```

**Pull request merged**